src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs

```diff
@@ -327,10 +327,19 @@ internal static void ApplyParameterInfo(this JsonNode schema, ApiParameterDescri
     /// <param name="createSchemaReferenceId">A delegate that generates the reference ID to create for a type.</param>
     internal static void ApplyPolymorphismOptions(this JsonNode schema, JsonSchemaExporterContext context, Func<JsonTypeInfo, string?> createSchemaReferenceId)
     {
-        // The `context.Path.Length == 0` check is used to ensure that we only apply the polymorphism options
+        // The `context.BaseTypeInfo == null` check is used to ensure that we only apply the polymorphism options
         // to the top-level schema and not to any nested schemas that are generated.
-        if (context.TypeInfo.PolymorphismOptions is { } polymorphismOptions && context.Path.Length == 0)
+        if (context.TypeInfo.PolymorphismOptions is { } polymorphismOptions && context.BaseTypeInfo == null)
         {
+            // System.Text.Json supports serializing to a non-abstract base class if no discriminator is provided.
+            // OpenAPI requires that all polymorphic sub-schemas have an associated discriminator. To maintain accuracy,
+            // with the actual implementation and comply with the limitations of the OpenAPI schema we do not emit
```

Comment:
The "actual implementation" being STJ?

Reply:
Yep, actual implementation here means what the STJ serializers do when the app is running.

Reply:
I would find it clearer to say "STJ" then.

Reply:
Updated.